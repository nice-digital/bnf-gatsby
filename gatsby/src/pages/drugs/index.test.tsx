import { render, waitFor, screen } from "@testing-library/react";

import DrugsIndexPage, { type DrugsIndexPageProps } from "./";

const props: DrugsIndexPageProps = {
	data: {
		allDrugs: {
			letters: [
				{
					letter: "a",
					links: [
						{
							title: "Abacavir",
							slug: "abacavir",
						},
						{
							title: "Abatacept",
							slug: "abatacept",
						},
					],
				},
				{
					letter: "b",
					links: [
						{
							title: "Bacillus Calmette-Guérin",
							slug: "bacillus-calmette-guerin",
						},
						{
							title: "Bacillus Calmette-Guérin",
							slug: "bacillus-calmette-guerin-vaccine",
						},
					],
				},
			],
		},
	},
};

describe("DrugsIndexPage", () => {
	it("should set page title", async () => {
		render(<DrugsIndexPage {...props} />);
		await waitFor(() => {
			expect(document.title).toStartWith("Drugs A to Z");
		});
	});

	it("should set page meta description", async () => {
		render(<DrugsIndexPage {...props} />);
		await waitFor(() => {
			expect(
				document
					// eslint-disable-next-line testing-library/no-node-access
					.querySelector("meta[name='description']")
					?.getAttribute("content")
			).toEqual("Browse the complete list of drugs, alphabetically.");
		});
	});

	it("should match snapshot for page body", () => {
		render(<DrugsIndexPage {...props} />);
		expect(screen.getByRole("main")).toMatchSnapshot();
	});
});
