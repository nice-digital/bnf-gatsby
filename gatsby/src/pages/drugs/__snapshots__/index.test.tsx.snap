// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrugsIndexPage should match snapshot for page body 1`] = `
<main>
  <div
    class="container"
  >
    <nav
      aria-label="Breadcrumbs"
      role="navigation"
    >
      <ol
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="https://www.nice.org.uk/"
          >
            NICE
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/"
          >
            BNF
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <span>
            Drugs A to Z
          </span>
        </li>
      </ol>
    </nav>
    <script
      type="application/ld+json"
    >
      {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "https://www.nice.org.uk/",
        "name": "NICE"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/",
        "name": "BNF"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "name": "Drugs A to Z"
      }
    }
  ]
}
    </script>
    <div
      class="page-header"
      id="content-start"
    >
      <h1
        class="page-header__heading"
      >
        Drugs
         A to Z
      </h1>
      <p
        class="page-header__lead"
      >
        Browse drug monographs by A to Z. Drug monographs describe the uses, doses, safety issues, medicinal forms and other considerations involved in the use of a drug.
      </p>
    </div>
    <ol
      class="alphabet  "
      id="a-to-z"
    >
      <li
        class="alphabet__letter "
      >
        <a
          href="#a"
        >
          A
        </a>
      </li>
      <li
        class="alphabet__letter "
      >
        <a
          href="#b"
        >
          B
        </a>
      </li>
    </ol>
    <ol
      class="a-z-list "
    >
      <li
        class="a-z-list__item "
      >
        <h2
          class="a-z-list__item-heading"
          id="a"
        >
          A
        </h2>
        <ol
          aria-labelledby="a"
          class="column-list column-list--boxed"
        >
          <li>
            <a
              href="/drugs/abacavir/"
            >
              Abacavir
            </a>
          </li>
          <li>
            <a
              href="/drugs/abatacept/"
            >
              Abatacept
            </a>
          </li>
        </ol>
      </li>
      <li
        class="a-z-list__item "
      >
        <h2
          class="a-z-list__item-heading"
          id="b"
        >
          B
        </h2>
        <ol
          aria-labelledby="b"
          class="column-list column-list--boxed"
        >
          <li>
            <a
              href="/drugs/bacillus-calmette-guerin/"
            >
              Bacillus Calmette-Guérin
            </a>
          </li>
          <li>
            <a
              href="/drugs/bacillus-calmette-guerin-vaccine/"
            >
              Bacillus Calmette-Guérin
            </a>
          </li>
        </ol>
      </li>
    </ol>
  </div>
</main>
`;
